src/EditorFeatures/VisualBasicTest/UseCompoundAssignment/UseCompoundAssignmentTests.vb

```diff
@@ -0,0 +1,341 @@
+        end sub
+    end class
+end namespace",
+"namespace NS
+    public class C
+        shared private a as integer
+
+        sub M()
+            NS.C.a += 10
+        end sub
+    end class
+end namespace")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestThroughBase() As Task
+            Await TestInRegularAndScriptAsync(
+"public class C
+    public a as integer
+end class
+
+public class D
+    inherits C
+    sub M()
+        mybase.a [||]= mybase.a + 10
+    end sub
+end class",
+"public class C
+    public a as integer
+end class
+
+public class D
+    inherits C
+    sub M()
+        mybase.a += 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestMultiAccess() As Task
+            Await TestInRegularAndScriptAsync(
+"public class C
+    public a as integer
+end class
+
+public class D
+    private c as C
+
+    sub M()
+        me.c.a [||]= me.c.a + 10
+    end sub
+end class",
+"public class C
+    public a as integer
+end class
+
+public class D
+    private c as C
+
+    sub M()
+        me.c.a += 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestNotOnProp() As Task
+            Await TestMissingAsync(
+"public class C
+    public property a as integer
+    end property
+
+    sub M()
+        a [||]= a + 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestNotOnUnboundSymbol() As Task
+            Await TestMissingAsync(
+"public class C
+    sub M()
+        a [||]= a + 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestNotWithSideEffects() As Task
+            Await TestMissingAsync(
+"public class C
+    private i as integer
+
+    function Goo() as C
+        return me
+    end function
+
+    sub M()
+        me.Goo().i [||]= me.Goo().i + 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestTrivia() As Task
+            Await TestInRegularAndScriptAsync(
+"public class C
+    sub M(a as integer)
+        // before
+        a [||]= a + 10 // after
+    end sub
+end class",
+"public class C
+    sub M(a as integer)
+        // before
+        a += 10 // after
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestFixAll() As Task
+            Await TestInRegularAndScriptAsync(
+"public class C
```

Comment:
It's very intentional.  It helps ensure that we can know that the change made to the file is not "too big".  That's because if we accidentally change too much, then we'll see VB change the capitalization of stuff we don't want changed.

In general, because VB is so aggressive with cleaning up touched code, we want to be very careful to make the minimal change possible.  Some features in the past even ended up marking the 'root' as needing ot be cleaned up, which could lead to massive edits to the user's code.

Reply:
Interesting. Makes sense

---
In reply to: [218628037](https://github.com/dotnet/roslyn/pull/29919#discussion_r218628037) [](ancestors = 218628037)